### ISSUE TITLE
[a11y]:DatePicker: Text contrast of 1.55 with its background is less than the WCAG AA minimum requirements for text of size 14px and weight of 400

### ISSUE THREAD

**Issue by yujinpingping** [carbon-design-system/carbon#15290]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

v10.56.0

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

Text contrast of 1.55 with its background is less than the WCAG AA minimum requirements for text of size 14px and weight of 400,
![image](https://github.com/carbon-design-system/carbon/assets/151118226/6c3f74ec-1646-4dc1-be35-2d305d40120e)

### WCAG 2.1 Violation

Contrast (Minimum) (Level AA)

### Reproduction/example

N/A

### Steps to reproduce

Go to Messaging https://dev.console.test.cloud.ibm.com/messaging
Click the Calendar

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by alisonjoseph** [comment]
Hi, disabled states aren't required to hit contrast requirements.
https://www.w3.org/WAI/WCAG21/Understanding/contrast-minimum.html
